p2p_wallet/Services/Repositories/BaseRepository/ItemRepository.swift

```diff
@@ -36,74 +35,61 @@ class ItemRepository<ItemType: Hashable>: ObservableObject {
     
     /// The request to retrieve data into repository
     /// - Returns: item
-    func createRequest() async throws -> ItemType {
+    internal func request() async throws -> ItemType {
         fatalError("Must override")
     }
     
     // MARK: - Actions
 
     /// Erase data and reset repository to its initial state
-    func flush() {
+    public func reset(autoFetch: Bool = true) {
         data = initialData
         state = .initializing
         error = nil
-    }
-    
-    /// Erase and reload all data
-    func reload() {
-        flush()
-        request()
-    }
-    
-    /// Refresh data without erasing current data
-    func refresh() {
-        request()
-    }
-    
-    /// Indicate if should fetch new data to prevent unwanted request
-    /// - Returns: should fetch new data
-    func shouldRequest() -> Bool {
-        true
+        
+        if autoFetch {
+            fetch()
+        }
     }
     
     /// Request data from outside to get new data
-    func request() {
-        // prevent unwanted request
-        guard shouldRequest() else {
-            return
-        }
+    public func fetch() {
+        // Ensure only one task at current moment
+        guard loadingTask == nil else { return }
```

Comment:
Sometimes we want to cancel the current task (reload), with this guarding, the task will not be cancelled when it is executing